### ISSUE TITLE
SR() cannot parse  'hypergeometric((3,), (...,)..)'

### ISSUE THREAD

**Issue by dimpase** [sagemath/sage#35572]
### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### Did you read the documentation and troubleshoot guide?

- [X] I have read the documentation and troubleshoot guide

### Environment

```markdown
Sage 9.8,  Sage 10.0.rc0
```

### Steps To Reproduce

```
sage: sage: h=hypergeometric([3/2,], [5/2, 3], -1/4*3^2)
sage: SR(str(h))
```

### Expected Behavior

outputs
```
hypergeometric((3/2,), (5/2, 3), -1/4*3^2)
```
or whatever the way to represent this is

### Actual Behavior

the parser chokes up, because internally `h==hypergeometric((3/2,), (5/2, 3), -1/4*3^2)` - the 1st two args are tuples, not lists, and this is also the format of `str(h)`:
```
---------------------------------------------------------------------------
SyntaxError                               Traceback (most recent call last)
[...]
File ~/work/software/sage/src/sage/misc/parser.pyx:1042, in sage.misc.parser.Parser.parse_error()
   1041 cdef parse_error(self, Tokenizer tokens, msg="Malformed expression"):
-> 1042     raise SyntaxError(msg, tokens.s, tokens.pos)
   1043 

SyntaxError: Mismatched parentheses
...
```

### Additional Information

reported in https://groups.google.com/d/msgid/sage-devel/eeec7521-c567-4870-8442-e3eed439b32dn%40googlegroups.com

Note that 
```
sage: SR('hypergeometric([3/2,], [5/2, 3], -1/4*3^2)')
```
works just fine.

One possible fix thus would be to add an extra `__repr__`  (or `_repr_` ?) for this and related functions.

**Comment by mkoeppe** [comment]
> One possible fix thus would be to add an extra `__repr__` (or `_repr_` ?) for this and related functions.

Yes, I would be in favor of fixing how hypergeometric_function expressions print. I think there is no good reason for Python's idiosyncratic tuple notation (...,) to appear in symbolic expressions. 

But I think it's a bit trickier than supplying a repr method.
The normalization of lists to tuples happens in `new_Expression_from_pyobject`.
Printing takes place under control of Pynac.
